### DISCUSSION TITLE
Token saving location

### DISCUSSION THREAD

**Discussion by avmsl** [homeassistant-apps/app-cloudflared#306] [Q&A] [answered]
Hi, at first think you for your plugin, which has been a blessing and made me aware of the existence of the cloudlare tunnels that wil help  me in many projects.
Going to my issue, I wonder how could I get the configuration script of the addon to fetch the token from another place. I pan to use a hashicorp vault to store all my HA secrets in the future (based on https://community.home-assistant.io/t/using-hashicorp-vault-to-store-secrets/454943), but for the moment I am trying to use the secrets.yaml as a middle point.
However, I could not get the cloudflared plugin to fetch a valid token from there using a similar coding to what I use in other addons, i.e.:
`tunnel_token= '!secrets cloudflare_token'` 
Thank you for your help!
António

**Comment by brenner-tobias** [comment]
I like this suggestion, but am right now not sure how the secrets file is working. I will have a look. Maybe @elcajon has some thoughts on this as well

**Comment by elcajon** [comment, marked as answer]
This works without additional configuration if the token is located in the /config/secrets.yaml file. However, `!secret` must be used instead of `!secrets` in the add-on configuration.

This is my working configuration:

<img width="649" alt="image" src="https://user-images.githubusercontent.com/52052494/229847448-d3f6047d-913d-4a06-aff3-b89e13d7b38d.png">

**Reply by brenner-tobias** [reply]
Thanks a lot for checking. @avmsl can you let us know if this works for you as well?

**Reply by akirayamamoto** [reply]
This is not working for me. I am getting the following error message: `Failed to save add-on configuration, Unknown secret 'cloudflared_tunnel_token' in Cloudflared (9074a9fa_cloudflared). Got {'external_hostname': '', 'additional_hosts': [], 'tunnel_token': '!secret cloudflared_tunnel_token'}`

The secret works in the Google Drive Backup addon though.

**Reply by akirayamamoto** [reply]
Actually, this is working fine. I just needed to restart the addon after updating my secrets file.
